From f3f58ea2b7dd334080633eba0c5c3ab87d66918e Mon Sep 17 00:00:00 2001
From: Meric Sentunali <merics@wirespring.com>
Date: Fri, 19 Jun 2020 18:37:27 +0300
Subject: [PATCH] fix memory leak when a custom type in rules does not match

---
 src/pdag.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/pdag.c b/src/pdag.c
index 44a3847..8a828e0 100644
--- a/src/pdag.c
+++ b/src/pdag.c
@@ -1442,6 +1442,10 @@ tryParser(npb_t *const __restrict__ npb,
 		LN_DBGPRINTF(dag->ctx, "called CUSTOM PARSER '%s', result %d, "
 			"offs %zd, *pParsed %zd", dag->ctx->type_pdags[prs->custTypeIdx].name, r, *offs, *pParsed);
 		*pParsed = npb->parsedTo - *offs;
+		if (r != 0) {
+			json_object_put(*value);
+			*value = NULL;
+		}
 		#ifdef	ADVANCED_STATS
 		es_addBuf(&npb->astats.exec_path, hdr, lenhdr);
 		es_addBuf(&npb->astats.exec_path, "[R:USR],", 8);
