diff --git a/configure.ac b/configure.ac
index 70360ce6f..925b4856c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2795,12 +2795,12 @@ AC_ARG_ENABLE(omotel,
          esac],
         [enable_omotel=no]
 )
-OMOTEL_HTTP_CFLAGS=""
-OMOTEL_HTTP_LIBS=""
 OMOTEL_PROTOBUF_CFLAGS=""
 OMOTEL_PROTOBUF_LIBS=""
 if test "x$enable_omotel" = "xyes"; then
-        PKG_CHECK_MODULES([OMOTEL_HTTP], [libcurl libfastjson])
+        if test "x${OMOTEL_HTTP_CFLAGS}${OMOTEL_HTTP_LIBS}" = "x"; then
+            PKG_CHECK_MODULES([OMOTEL_HTTP], [libcurl libfastjson])
+        fi
         PKG_CHECK_MODULES([OMOTEL_PROTOBUF], [libprotobuf-c >= 1.0.0])
         AC_CHECK_PROG([PROTOC_C], [protoc-c], [protoc-c])
         if test "x$PROTOC_C" = "x"; then
