--- a/plugins/fmpcre/fmpcre.c
+++ b/plugins/fmpcre/fmpcre.c
@@ -22,7 +22,7 @@

 MODULE_TYPE_FUNCTION
 MODULE_TYPE_NOKEEP
-DEF_FMOD_STATIC_DATA
+DEF_FMOD_STATIC_DATA;

 struct pcre_state {
     pcre *re;
@@ -107,6 +107,6 @@ BEGINqueryEtryPt
 ENDqueryEtryPt

 BEGINmodInit()
-    CODESTARTmodInit *ipIFVersProvided = CURR_MOD_IF_VERSION;
+    CODESTARTmodInit; *ipIFVersProvided = CURR_MOD_IF_VERSION;
     CODEmodInit_QueryRegCFSLineHdlr dbgprintf("rsyslog fmpcre init called, compiled with version %s\n", VERSION);
 ENDmodInit
