diff --git a/tests/Makefile.am b/tests/Makefile.am
index f4eead9e3..add8f9c3c 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -1817,6 +1817,9 @@ CLEANFILES+= \
 
 pkglib_LTLIBRARIES =
 
+# TODO: reenable TESTRUNS = rt_init rscript
+if ENABLE_TESTBENCH
+
 pkglib_LTLIBRARIES += liboverride_gethostname.la
 liboverride_gethostname_la_SOURCES = override_gethostname.c
 liboverride_gethostname_la_CFLAGS =
@@ -1832,8 +1835,6 @@ liboverride_getaddrinfo_la_SOURCES = override_getaddrinfo.c
 liboverride_getaddrinfo_la_CFLAGS =
 liboverride_getaddrinfo_la_LDFLAGS = -avoid-version -shared
 
-# TODO: reenable TESTRUNS = rt_init rscript
-if ENABLE_TESTBENCH
 check_PROGRAMS = $(TESTRUNS) ourtail tcpflood chkseq msleep randomgen \
 	diagtalker uxsockrcvr syslog_caller inputfilegen minitcpsrv \
 	omrelp_dflt_port \
