--- rsyslog-8.2402.0.ori/plugins/imdtls/Makefile.am	2024-02-26 08:24:02.000000000 +0000
+++ rsyslog-8.2402.0/plugins/imdtls/Makefile.am	2024-02-29 15:22:48.649238081 +0000
@@ -2,5 +2,5 @@
 imdtls_la_DEPENDENCIES = ../../runtime/lmnsd_ossl.la
 imdtls_la_SOURCES = imdtls.c
 imdtls_la_CPPFLAGS = -I$(top_srcdir) $(PTHREADS_CFLAGS) $(RSRT_CFLAGS) $(OPENSSL_CFLAGS)
-imdtls_la_LDFLAGS = -module -avoid-version
-imdtls_la_LIBADD = $(OPENSSL_LIBS) ../../runtime/lmnsd_ossl.la
+imdtls_la_LDFLAGS = -module -avoid-version -L../../runtime/.libs -l:lmnsd_ossl.so -Wl,-R,'$$ORIGIN/'
+imdtls_la_LIBADD = $(OPENSSL_LIBS)
diff -ru ../../../rsyslog-8.2402.0.ori/plugins/omdtls/Makefile.am ../../../rsyslog-8.2402.0/plugins/omdtls/Makefile.am
--- rsyslog-8.2402.0.ori/plugins/omdtls/Makefile.am	2024-02-26 08:24:02.000000000 +0000
+++ rsyslog-8.2402.0/plugins/omdtls/Makefile.am	2024-02-29 15:23:02.104298548 +0000
@@ -2,5 +2,5 @@
 omdtls_la_DEPENDENCIES = ../../runtime/lmnsd_ossl.la
 omdtls_la_SOURCES = omdtls.c
 omdtls_la_CPPFLAGS = -I$(top_srcdir) $(PTHREADS_CFLAGS) $(RSRT_CFLAGS) $(OPENSSL_CFLAGS)
-omdtls_la_LDFLAGS = -module -avoid-version
-omdtls_la_LIBADD = $(OPENSSL_LIBS) ../../runtime/lmnsd_ossl.la
+omdtls_la_LDFLAGS = -module -avoid-version -L../../runtime/.libs -l:lmnsd_ossl.so -Wl,-R,'$$ORIGIN/'
+omdtls_la_LIBADD = $(OPENSSL_LIBS)
