--- rsyslog.service.in	2020-06-22 14:57:15.000000000 +0200
+++ rsyslog.service	2020-07-05 17:10:55.166026973 +0200
@@ -4,11 +4,13 @@
 Wants=network.target network-online.target
 After=network.target network-online.target
 Documentation=man:rsyslogd(8)
-Documentation=http://www.rsyslog.com/doc/
+Documentation=https://www.rsyslog.com/doc/
 
 [Service]
 Type=notify
-ExecStart=@sbindir@/rsyslogd -n -iNONE
+EnvironmentFile=-/etc/sysconfig/rsyslog
+ExecStart=/usr/sbin/rsyslogd -n $SYSLOGD_OPTIONS
+UMask=0066
 StandardOutput=null
 Restart=on-failure
 
