From 74345a12854f1181f1c32ba52a3d279ad8fe6d87 Mon Sep 17 00:00:00 2001
From: Julien Thomas <jthomas@zenetys.com>
Date: Sun, 16 Aug 2020 07:16:38 +0200
Subject: [PATCH 1/4] msg: expose function jsonDeepCopy()

---
 runtime/msg.c | 4 ++--
 runtime/msg.h | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/runtime/msg.c b/runtime/msg.c
index 6468a0d55..b4235e4ba 100644
--- a/runtime/msg.c
+++ b/runtime/msg.c
@@ -423,7 +423,7 @@ static int getAPPNAMELen(smsg_t * const pM, sbool bLockMutex);
 static rsRetVal jsonPathFindParent(struct json_object *jroot, uchar *name, uchar *leaf,
 	struct json_object **parent, int bCreate);
 static uchar * jsonPathGetLeaf(uchar *name, int lenName);
-static struct json_object *jsonDeepCopy(struct json_object *src);
+struct json_object *jsonDeepCopy(struct json_object *src);
 static json_bool jsonVarExtract(struct json_object* root, const char *key, struct json_object **value);
 void getRawMsgAfterPRI(smsg_t * const pM, uchar **pBuf, int *piLen);
 
@@ -5075,7 +5075,7 @@ finalize_it:
 	RETiRet;
 }
 
-static struct json_object *
+struct json_object *
 jsonDeepCopy(struct json_object *src)
 {
 	struct json_object *dst = NULL, *json;
diff --git a/runtime/msg.h b/runtime/msg.h
index 32b155910..00d782dc5 100644
--- a/runtime/msg.h
+++ b/runtime/msg.h
@@ -239,6 +239,7 @@ unsigned short *pbMustBeFreed);
 rsRetVal msgSetJSONFromVar(smsg_t *pMsg, uchar *varname, struct svar *var, int force_reset);
 rsRetVal msgDelJSON(smsg_t *pMsg, uchar *varname);
 rsRetVal jsonFind(struct json_object *jroot, msgPropDescr_t *pProp, struct json_object **jsonres);
+struct json_object *jsonDeepCopy(struct json_object *src);
 
 rsRetVal msgPropDescrFill(msgPropDescr_t *pProp, uchar *name, int nameLen);
 void msgPropDescrDestruct(msgPropDescr_t *pProp);
-- 
2.21.1

